docs/protocol.md

```diff
@@ -1,211 +1,370 @@
+| -------------------------- | ------------------------------------------------ | ---------------------------- |
+| $`(skS_{dh}, pkS_{dh})`$   | $`(S_{dh,sk}, S_{dh,pk})`$                       | $`(J_{dh,sk}, J_{dh,pk})`$   |
+| $`(skR_{dh}, pkR_{dh})`$   | $`(J_{\textbf{edh},sk}, J_{\textbf{edh},pk})`$   | $`(S_{dh,sk}, S_{dh,pk})`$   |
+| $`(skR_{kem}, pkR_{kem})`$ | $`(J_{\textbf{ekem},sk}, J_{\textbf{ekem},pk})`$ | $`(S_{kem,sk}, S_{kem,pk})`$ |
+
+> For messages sent from a source to a journalist, the source is identified by
+> $`S_{dh,pk}`$ and utilizes the [bolded] ephemeral keys $`J_{edh,pk}`$ and
+> $`J_{ekem,pk}`$ to encrypt its message. The journalist, in turn, authenticates
+> itself using the new long-term key $`J_{dh,pk}`$ and relies on the source's
+> long-term keys $`S_{dh,pk}`$ and $`S_{kem,pk}`$ to encrypt messages back to the
+> source securely. (Maier §5.2)
+
+## Setup
+
+### 1. FPF
+
+| FPF                                                      |
+| -------------------------------------------------------- |
+| $`(FPF_{sig,sk}, FPF_{sig,pk}) \gets^{\$} \text{Gen}()`$ |
+
+The server, the journalist client, and the source client SHOULD be built with
+$FPF_{sig,pk}$ pinned.[^1]
+
+### 2. Newsroom
+
+| Newsroom                                               |                                 | FPF                                                                |
+| ------------------------------------------------------ | ------------------------------- | ------------------------------------------------------------------ |
+| $`(NR_{sig,sk}, NR_{sig,pk}) \gets^{\$} \text{Gen}()`$ |                                 |                                                                    |
+|                                                        | $`\longrightarrow NR_{sig,pk}`$ | Verify manually                                                    |
+|                                                        |                                 | $`\sigma^{FPF} \gets^{\$} \text{Sign}(FPF_{sig,sk}, NR_{sig,pk})`$ |
+|                                                        | $`\sigma^{FPF} \longleftarrow`$ |
+
+The server MUST be deployed with $NR_{sig,pk}$ pinned. The server MAY be
+deployed with $\sigma^{FPF}$ pinned.[^1]
+
+### 3. Journalist
+
+#### 3.1. Enrollment
+
+| Journalist                                               |                                                         | Newsroom                                                                                   |
+| -------------------------------------------------------- | ------------------------------------------------------- | ------------------------------------------------------------------------------------------ |
+| $`(J_{sig,sk}, J_{sig,pk}) \gets^{\$} \text{Gen}()`$     |                                                         |                                                                                            |
```

Comment:
so in this part of the protocol the first invocation of the `Gen` function for the signing key should be the keygen algorithm of the signature scheme, and the `Gen` function for the others is just 
1) generate a random group element (i.e. $x \leftarrow \mathbb{Z}_p$), then 
2) do DH with the standard generator, i.e. $X = g^x$ or I guess $DH(g, x)$ in the nomenclature here? 

is this a correct understanding?